### PR TITLE
Kogito 1020 : VSCode - BPMN modeler support variables tagging

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin can you review this PR?

Ready for Review
Known issues that I will be fixing while this PR is reviewed:
-When loading a new diagram that has tags, when you add other tags the previous tags are removed (Only one time) - FIXED NOW

-Additional tests are needed to test the logic of the overlay and adding of the badges/labels and edge cases - Fixed Some Tests / ADDED
-Currently when saving it separates tags with semicolon ; as it caused conflicts with the separation of variables, will fix - FIXED NOW
-Business Central needs to be tested to ensure that it can save and reload FIXED ISSUES / TESTED
-Currently when saving CData is skipped Fixed by Kirill on another PR

****Note : Due to the last refactor, the widget does not work properly in regards to events, will fix tomorrow. Firing up the builds to check Sonar Cloud Issues -- FIXED NOW

### CODE REVIEW

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java [hunk 1]
```diff
@@ -31,7 +31,20 @@
     public static final MetadataTypeDefinition<Boolean> autoStart = new BooleanElement("customAutoStart", false);
     public static final MetadataTypeDefinition<Boolean> autoConnectionSource = new BooleanElement("isAutoConnection.source", false);
     public static final MetadataTypeDefinition<Boolean> autoConnectionTarget = new BooleanElement("isAutoConnection.target", false);
-    public static final MetadataTypeDefinition<Boolean> customKPI = new BooleanElement("customKPI", false);
+    public static final MetadataTypeDefinition<String> customTags = new StringElement("customTags", "[]") {
+        // Need to use an intermediary character since Variables depend on comma separators
+        @Override
+        public String getValue(BaseElement element) {
+            return getStringValue(element)
+                    .orElse(getDefaultValue()).replace(",", ";");
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java]: not sure if makes sense, but shouldn't it be a `replaceAll` instead of a `replace`?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java [hunk 2]
```diff
@@ -31,7 +31,20 @@
...
+        }
+
+        @Override
+        public void setValue(BaseElement element, String value) {
+            value = value.replace(";", ",");
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java]: not sure if makes sense, but shouldn't it be a `replaceAll` instead of a `replace`?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html [hunk 3]
```diff
@@ -16,8 +16,10 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
+
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html]: This `html template` file represents a "fragment" of the whole `html` document, so there is no need to declare neither the `head` and `body` tags, could you please remove all those? The html template should just start on the `<div id="widget"...`.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html [hunk 4]
```diff
@@ -30,40 +32,61 @@
...
-                            <th data-field="datatypeth" style="width:46%">
+                            <th data-field="datatypeth" style="width:43%">
                                 Data Type
                             </th>
 
-                            <th data-field="kpith" style="width:4%">
-                                KPI
+                            <th style="width:22%" data-field="tagsth">
+                                Tags
+                                <i class="fa fa-info-circle" title = "Add Predefined or Custom Tags to Variables"/>
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html]: no i18n

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html [hunk 5]
```diff
@@ -30,40 +32,61 @@
...
                                 </select>
                                 <input type="text" class="form-control" data-field="customDataType"/>
                             </td>
 
-                            <td style="width:4%" data-field="kpiTD">
-                                <input type="checkbox" style="margin-left:15px; margin-top:10px" data-field="kpi"/>
+                            <td style="width:22%" data-field="tagsTD">
+                                <div class = "containerDiv" title = "Tags">
+                                    <a class = "tagAnchor tagsSVG" data-field="variable-tags-settings" data-dismiss="onblur" data-placement="bottom" data-toggle="popover" data-html="true" data-content="OverlayContent">
+                                        <span data-field ="tagCount"  title = "These are the tags included " class = "tagSpan"></span>
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html]: no i18n

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html [hunk 6]
```diff
@@ -30,40 +32,61 @@
...
+                                    </a>
+                                </div>
+
+                                <div style="display:none;"   data-field = "tags-div" class = "tagsDiv">
+
+                                    <button type="button" class ="close" data-field="closeButton">x</button>
+
+                                    <h3>Variable tags</h3>
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html]: no i18n

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html [hunk 7]
```diff
@@ -30,40 +32,61 @@
...
+
+                                    <select data-field="defaultTagNames" class="form-control selectionWidget" style = "width: 150px;"/>
+                                    <input class="form-control selectionWidget" data-field="customTagName" style = "width: 150px;" type="text"/>
+
+                                    <span class="glyphicon glyphicon-ok check-mark acceptButton" title = "Add Tag" data-field="acceptButton" ></span>
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html]: no i18n

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java [hunk 8]
```diff
@@ -78,13 +78,13 @@
     private DataTypeNamesService clientDataTypesService;
 
     @DataField
-    protected TableCellElement kpith = Document.get().createTHElement();
+    protected TableCellElement tagsth = Document.get().createTHElement();
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java]: could it be `private` access?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java [hunk 9]
```diff
@@ -78,13 +78,13 @@
...
 
     List<String> dataTypes;
     List<String> dataTypeDisplayNames;
     boolean readOnly = false;
 
-    private boolean kpiDisabled = false;
+    protected boolean tagsDisabled = false;
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java]: could it be `private` access?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java [hunk 10]
```diff
@@ -30,7 +30,21 @@
     public static final MetadataTypeDefinition<Boolean> autoStart = new BooleanElement("customAutoStart", false);
     public static final MetadataTypeDefinition<Boolean> autoConnectionSource = new BooleanElement("isAutoConnection.source", false);
     public static final MetadataTypeDefinition<Boolean> autoConnectionTarget = new BooleanElement("isAutoConnection.target", false);
-    public static final MetadataTypeDefinition<Boolean> customKPI = new BooleanElement("customKPI", false);
+    public static final MetadataTypeDefinition<String> customTags = new StringElement("customTags", "[]") {
+        // Need to use an intermediary character since Variables depend on comma separators
+        @Override
+        public String getValue(BaseElement element) {
+            return getStringValue(element)
+                    .orElse(getDefaultValue()).replace(",", ";");
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java]: `replace` or `replaceAll` here too?'

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java [hunk 11]
```diff
@@ -30,7 +30,21 @@
...
+        }
+
+        @Override
+        public void setValue(BaseElement element, String value) {
+            value = value.replace(";", ",");
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java]: `replace` or `replaceAll` here too?'

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariableListItemWidgetViewImpl.java [hunk 12]
```diff
@@ -58,7 +77,7 @@
  * they use a combination of ListBox and TextBox to implement a drop-down combo
  * to hold the values.
  */
-@Templated("VariablesEditorWidget.html#variableRow")
+@Templated(value = "VariablesEditorWidget.html#variableRow", stylesheet = "VariablesEditorWidget.css")
 public class VariableListItemWidgetViewImpl implements VariableListItemWidgetView,
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariableListItemWidgetViewImpl.java]: In general I see this `VariableListItemWidgetViewImpl` class, which is a view implementatio, plenty of logic. Not sure in the already existing stuff, but at least looking at all below new methods.

The ideal should be having no logic on the view implementations, this way the classes are easier to test and most important, it results easy to manage the code and move to other (future ) implementation of a view, by reusing the same presenter classes and unit tests.

So it does not worth adding a comment for each of the conditional or looping expressions below in this class, just give it a second look now, try to see wihch logics can be moved to presenter, and then we can do second review iteration for this class. So please, could you try to refactor this a bit and try to decouple the logic as much as possible from the view?

PS: Also could you please use `private` access modifier for the fields in the class? mostly are `protected`, not sure if for some reaon...

Thanks!

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java (hunk 1 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java:39]: Hi @romartin, I am pretty sure replace works by replacing all strings, replaceAll as per the documentation replaces all as well but working using a Regex, let me know if I should change

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java (hunk 2 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java:44]: Same as above @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html (hunk 5 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html:68]: Added @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html (hunk 6 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html:76]: Added @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html (hunk 7 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidget.html:81]: Added @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java (hunk 8 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java:81]: Fixed @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java (hunk 9 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariablesEditorWidgetViewImpl.java:87]: Fixed @romartin

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java (hunk 10 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java:38]: Hi @romartin comment from above

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java (hunk 11 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java:43]: Hi @romartin comment from above

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariableListItemWidgetViewImpl.java (hunk 12 above)
Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/forms/fields/variablesEditor/VariableListItemWidgetViewImpl.java:81]: Hi @romartin refactored some things, let me know if ok

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Hi @romartin ready for re-review, I refactored a bit more

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Missing a single bug on the tags dropdown - actuallly shows all the data types on the dropdown

**Comment by inodeman** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java (hunk 1 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java:39]: Hey @inodeman 

No worries, maybe I'm also wrong hehe, let's try to figure it out together :) 

As far as I remember it's replacling all occurrences but in case you're using a regular expression, but here you're NOT using a regexp. So instead, you have to move this string to a regexp or use the replaceAll. Also think that even using the Java API, we're using same code for the client side marshallers, and that means that it will be transpiled to Javascript, where the  String API differs a bit. See for example this [post for the Javascript API](https://dmitripavlutin.com/replace-all-string-occurrences-javascript/), it mentions that String arguments are not considered regexp, so not replaccing all occurrences.

Sooo I think the best way to do this is by really testing both backend and server side implementations (around that `replace` method usages) by both unit testing and also testing the runtimes (BC and kogito). This way we'll be really sure about how each implementation works and if we're properly replacing ALL occurrences too.

Can you please do this quick exercice and try to ensure it at 100%? You think it makes sense?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java (hunk 2 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/customproperties/CustomElement.java:44]: same as above too :)

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java (hunk 11 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/customproperties/CustomElement.java:43]: same as above too but here we're talking about Javascript, not as in the backend marshallers

**Comment by romartin** [comment]
Hey @inodeman 

Please lemme know once above comments on the code are clarified and the bug on dropdowns fixed, so I can give the last look at this :+1: 

Thanks!

**Comment by lazarotti** [comment]
I just reviewed the bug about the dropdown and I couldn' t reproduce it anymore, so I am approving this.
@romartin aee that I strongly recommend just after the dev preview we have the enhancements/fixes for https://issues.redhat.com/browse/KOGITO-1730 and  https://issues.redhat.com/browse/KOGITO-1731

**Comment by LuboTerifaj** [comment]
Hey @inodeman , @romartin ,

the PR removes also KPI for process variables. If you import the process created in Stunner editor from master, the KPI values (true or false) are not processed and the data is lost.

**Comment by inodeman** [comment]
Hi @LuboTerifaj fixed the issue about Global/Case Variables about error in pop up for not getting DataTypes, about the other issues

- Global/Case Variables should not have Tags, only process variables

- Issue about UX Improvements should be fixed in ticket
https://issues.redhat.com/browse/KOGITO-1730

- Issue about Edit will be fixed  also in since Editing option will be removed
https://issues.redhat.com/browse/KOGITO-1730

Also, this new change will also be enhanced
https://issues.redhat.com/browse/KOGITO-1731

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by lazarotti** [comment]
> Hey @inodeman , @romartin ,
> 
> the PR removes also KPI for process variables. If you import the process created in Stunner editor from master, the KPI values (true or false) are not processed and the data is lost.

@LuboTerifaj I discussed about this during the last meeting. The engine had no implementation about the KPI and it was imported to Stunner from legacy to allow some further implementation in the engine side. We had only one customer that asked for something that could allow them to develop their own solution, but that was not supported by say. The tagging mechanism provides now official hooks to users to implement what they want, so KPI as custom tag found a correct place that could be extended in some supported way by them. I asked Martin (CEE) about this and Support was fine about dropping the column. As we didn't have implementation nor listeners for the KPI previously, we do not need to take care of processes with this previously set

**Comment by lazarotti** [comment]
> Hi @LuboTerifaj fixed the issue about Global/Case Variables about error in pop up for not getting DataTypes, about the other issues
> 
> * Global/Case Variables should not have Tags, only process variables
@inodeman that is not correct, the feature is also about Cases Variable. You are right about Globals, but for sure it must work for cases

**Comment by romartin** [comment]
yeah @LuboTerifaj  in fact we discussed this on this weekly meeting as far as I remember, the KPI information will be lost, but that was acceptable for our customers (in fact just a single custom I guess is using it). That should not be an issue IMO, we'll not provide support for importing old KPI information anyway, we'll just rely on tags from now on.

Thx!

**Comment by inodeman** [comment]
Jenkins execute full downstream build

**Comment by inodeman** [comment]
Jenkins retest please

**Comment by lazarotti** [comment]
@LuboTerifaj, this PR is a blocker for Kogito dev preview what today is the cut-off. I and Roger reviewed it and it seems fine for the merge.
See that it can be reverted or fixed/improved at any moment if you see any problem in the product platform. I recommend to merge it to unblock kogito dev preview. and I invite you @LuboTerifaj to next week to keep the review to see if there is any Jira that should be opened

**Comment by romartin** [comment]
Ok @LuboTerifaj , let's report that issues for checking during next week. On the other hand, as it's an specific case, it should not block the feature, so we'll proceed to merging.

Thanks all guys for the great effort!

**Comment by romartin** [comment]
The test failed is not related to this PR: `org.kie.bc.DeploymentIT.org.kie.bc.DeploymentIT`

Merging.

**Comment by inodeman** [comment]
Hi @LuboTerifaj I reran everything with latest and could not reproduce issues mentioned by @LuboTerifaj , also tested Business Central war from builds on this PR and Case Variables worked fine, lets check next week to make sure it is issues

image
